File: library/src/test/java/com/stealthcopter/networktools/ARPInfoTest.java
package com.stealthcopter.networktools;

import org.junit.Test;

import static junit.framework.Assert.assertNull;

/**
 * To work on unit tests, switch the Test Artifact in the Build Variants view.
 */
public class ARPInfoTest {

    @Test
    public void nullIPsandMacsReturnNull() {
        assertNull(ARPInfo.getMACFromIPAddress(null));
        assertNull(ARPInfo.getIPAddressFromMAC(null));
    }

    @Test(expected = IllegalArgumentException.class)
    public void testIllegalArgumentThrownOnInvalidMACaddress() {
        ARPInfo.getIPAddressFromMAC("00:00:00:xx");
    }
}

File: library/src/test/java/com/stealthcopter/networktools/ARPInfoTest.kt
package com.stealthcopter.networktools

import junit.framework.Assert
import org.junit.Test

/**
 * To work on unit tests, switch the Test Artifact in the Build Variants view.
 */
class ARPInfoTest {
    @Test
    fun nullIPsandMacsReturnNull() {
        Assert.assertNull(ARPInfo.getMACFromIPAddress(null))
        Assert.assertNull(ARPInfo.getIPAddressFromMAC(null))
    }

    @Test(expected = IllegalArgumentException::class)
    fun testIllegalArgumentThrownOnInvalidMACaddress() {
        ARPInfo.getIPAddressFromMAC("00:00:00:xx")
    }
}